cola/gitcfg.py

```diff
@@ -204,14 +206,14 @@ def update(self):
 
         self.notify_observers(self.message_updated)
 
-    def read_config(self, path):
+    def read_config(self, path, *opt_args):
         """Return git config data from a path as a dictionary."""
 
         if BUILTIN_READER:
             return self._read_config_file(path)
 
         dest = {}
-        args = ('--null', '--file', path, '--list')
+        args = ('--null', '--file', path, '--list') + opt_args
```

Comment:
Thanks for adding this. I think we would always want to use `--includes`, even for system and repo configs, so it doesn't seem worth making this optional at the `read_config(...)` level. In other words, can be just add the logic here rather than having the callers worry about it?

Also, this feature was added in git 1.7.10. Can you please add a `config-includes` entry to `cola/version.py` and then have a runtime check here where we check `if version.check_git(context, 'config-includes'): ...` in order to decide whether or not to use the flag? That'll make the code compatible with older versions.

BTW, the context is passed to the gitcfg constructor but we don't hold onto it. You'll need to store it in a member variable in the constructor (eg. `self._context = context`) so that we can use it to query git here.

Reply:
I've made the change, grateful for your feedback.